cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -380,9 +390,66 @@ public void testQueryMetrics() throws Exception {
         "&groupBy=namespace,flowlet&start=" + start + "&end=" + end, groupByResult);
   }
 
+
   @Test
-  public void testQueryMetricsWithTags() throws Exception {
+  public void testTimeRangeQueryBatch() throws Exception {
+    // note: times are in seconds, hence "divide by 1000";
+    long start = (emitTs - 60 * 1000) / 1000;
+    long end = (emitTs + 60 * 1000) / 1000;
 
+    verifyRangeQueryResult(
+      "/v3/metrics/query?" + getTags("yourspace", "WCount1", "WCounter", "splitter") +
+        "&metric=system.reads&start=" + start + "&end="
+        + end, 2, 3);
+
+    JsonObject q1 = getSingleQueryJson(ImmutableMap.of("namespace", "yourspace", "app", "WCount1", "flow", "WCounter",
```

Comment:
please, in tests, add a comment what you are testing and how. So that it is easy to read - why 2 queries, etc.